### PR TITLE
add scdoc documentation tool

### PR DESCRIPTION
**Pull request by aacebedo**
Specify library name and version:  **scdoc/1.11.12**

I am not the author of the tool but I need it for some of my projects.

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/scdoc/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+class ScdocInstallerConan(ConanFile):
+    name = "scdoc"
+    description = "scdoc is a simple man page generator for POSIX systems written in C99."
+    topics = ("scdoc", "documentation")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://git.sr.ht/~sircmpwn/scdoc"
+    license = "Unknown"
+    settings = "os", "arch", "compiler", "build_type"
+    _autotools = None
+    no_copy_source = True
```
Comment (uilianries) [inline recipes/scdoc/all/conanfile.py:17]:
```suggestion
```

Only for header-only or very big packages like Boost C++

File: recipes/scdoc/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        with tools.chdir(os.path.join(self.source_folder, self._source_subfolder)):
+            autotools.make()
+            autotools.make(target="check")
```
Comment (uilianries) [inline recipes/scdoc/all/conanfile.py:52]:
```suggestion
```

We don't run unit tests or build checks.

File: recipes/scdoc/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        self.env_info.PATH.append(scdoc_root)
+        self._chmod_plus_x(os.path.join(scdoc_root, "scdoc"))
+
+    def validate(self):
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration(
+                "Builds aren't supported on Macos")
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "Builds aren't supported on Windows")
```
Comment (uilianries) [inline recipes/scdoc/all/conanfile.py:77]:
```suggestion
        if self.settings.os in ["Macos", "Windows"]:
            raise ConanInvalidConfiguration(
                f"Builds aren't supported on {self.settings.os}")
```

File: recipes/scdoc/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from conans import ConanFile, Meson, tools
```
Comment (uilianries) [inline recipes/scdoc/all/test_package/conanfile.py:1]:
```suggestion
from conans import ConanFile, tools
```

Unused

File: recipes/scdoc/all/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+import os
+
+
+class TestPackageConan(ConanFile):
+
+    def test(self):
+        self.run(
+            f"scdoc < {os.path.join(self.source_folder,'test_package.1.scd')}", run_environment=True)
```
Comment (uilianries) [inline recipes/scdoc/all/test_package/conanfile.py:9]:
```suggestion
         if not tools.cross_building(self):
            self.run(
                f"scdoc < {os.path.join(self.source_folder,'test_package.1.scd')}", run_environment=True)
```

Don't need to run when cross compiled, the binary won't run.

File: recipes/scdoc/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.43.0"
```
Comment (uilianries) [inline recipes/scdoc/all/conanfile.py:5]:
```suggestion
required_conan_version = ">=1.33.0"
```

I see no new features which require 1.43 here.

File: recipes/scdoc/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        with tools.chdir(os.path.join(self.source_folder, self._source_subfolder)):
```
Comment (uilianries) [inline recipes/scdoc/all/conanfile.py:50]:
```suggestion
        with tools.chdir(self._source_subfolder):
```

Without `no_copy_source`

File: recipes/scdoc/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+            autotools.make()
+            autotools.make(target="check")
+
+    def package(self):
+        autotools = self._configure_autotools()
+        with tools.chdir(os.path.join(self.source_folder, self._source_subfolder)):
```
Comment (uilianries) [inline recipes/scdoc/all/conanfile.py:56]:
```suggestion
        with tools.chdir(self._source_subfolder):
```

Without `no_copy_source`

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution, please, take a look on my review.

File: recipes/scdoc/all/conanfile.py (hunk 6 above)
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:5]: corrected

File: recipes/scdoc/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.43.0"
+
+
+class ScdocInstallerConan(ConanFile):
+    name = "scdoc"
+    description = "scdoc is a simple man page generator for POSIX systems written in C99."
+    topics = ("scdoc", "documentation")
```
[the remark this answers is not in the document]
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:11]: corrected

File: recipes/scdoc/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://git.sr.ht/~sircmpwn/scdoc"
+    license = "Unknown"
```
[the remark this answers is not in the document]
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:14]: Didn't know it was an MIT as it did not mentionned it.
Corrected

File: recipes/scdoc/all/conanfile.py (hunk 1 above)
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:17]: corrected

File: recipes/scdoc/all/conanfile.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/scdoc/all/conanfile.py:14]: Usually they don't mention, you need to read and compare with open source licenses.

File: recipes/scdoc/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    settings = "os", "arch", "compiler", "build_type"
+    _autotools = None
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def build_requirements(self):
+        self.build_requires("make/[>=4.3.0]")
```
[the remark this answers is not in the document]
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:24]: There is a comment here https://github.com/conan-io/conan-center-index/pull/8588#discussion_r776962846  indicating ranges may be considered for build dependencies. Is it confirmed ?

File: recipes/scdoc/all/conanfile.py (hunk 2 above)
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:52]: corrected

File: recipes/scdoc/all/test_package/conanfile.py (hunk 4 above)
Comment (aacebedo) [reply recipes/scdoc/all/test_package/conanfile.py:1]: corrected

File: recipes/scdoc/all/test_package/conanfile.py (hunk 5 above)
Comment (aacebedo) [reply recipes/scdoc/all/test_package/conanfile.py:9]: corrected

File: recipes/scdoc/all/conanfile.py (hunk 7 above)
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:50]: corrected

File: recipes/scdoc/all/conanfile.py (hunk 8 above)
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:56]: corrected

File: recipes/scdoc/all/conanfile.py (hunk 3 above)
Comment (aacebedo) [reply recipes/scdoc/all/conanfile.py:77]: corrected

**Review by uilianries — DISMISSED** [review]
LGTM Thank you!

**Comment by aacebedo** [comment]
> LGTM Thank you!

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by prince-chrismc — DISMISSED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Why did you rebase and force push?

Your PR was ready to be merged 😱 I opened a PR since the docs were not clear https://github.com/conan-io/conan-center-index/pull/8797

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

**Review by prince-chrismc — APPROVED** [review]

**Comment by aacebedo** [comment]
I am terribly sorry :(

I force pushed because I identified an issue with the pkg-config file generated by the recipe (It did not include the same info as the one packaged with the source preventing it to be used by other packages to use it correctly).

Shall I use fixup commits instead to apply modifications during a review ?

**Comment by prince-chrismc** [comment]
That's absolutely alright. Thanks for catching the issue!

Please make new commits 🙏 it's much easier for us to review that way.

File: recipes/scdoc/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.libdirs = []
+
+        scdoc_root = os.path.join(self.package_folder, "bin")
+        self.output.info(
+            "Appending PATH environment variable: {}".format(scdoc_root))
+        self.env_info.PATH.append(scdoc_root)
+        self._chmod_plus_x(os.path.join(scdoc_root, "scdoc"))
```
Comment (uilianries) [inline recipes/scdoc/all/conanfile.py:67]: We usually do it when packaging, but I think you want to be sure the user will have permission to execute the binary.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Pull request merged**